### DISCUSSION TITLE
Push notifications from server

### DISCUSSION THREAD

**Discussion by eatenpancreas** [tauri-apps/tauri#11180] [Q&A]
Hi, I'm working on a Tauri app for IOS and Android, what would i need on the Tauri side to enable me to send push notifications from a server to my phone. If it's possible I'd preferably use some existing plugins

Thanks in advance!

**Comment by ahqsoftwares** [comment]
It is possible only if you write an Android Service in Kotlin. There is most probably no rust apis or tauri plugins

**Reply by nunnally** [reply]
But Is there a way to include that inside my tauri app?

**Reply by eatenpancreas** [reply]
thanks for this. Worked for me. IOS is a different beast though

**Reply by eatenpancreas** [reply]
> But Is there a way to include that inside my tauri app?

Not that i know of, i don't think it's possible since the service exceeds the runtime of the rust environment (Eg: when it's off, a service still runs. Tauri doesn't)

**Reply by ahqsoftwares** [reply]
It's possible, using kotlin demons, they run independent of tauri and is handled by android to trigger even when the app is not running

**Reply by osdiab** [reply]
Someone appears to be trying to contribute something for this, though it’s been a couple months since any updates on that.

https://github.com/tauri-apps/tauri/issues/11651
